kbase-extension/static/kbase/js/widgets/appWidgets2/input/dynamicDropdownInput.js

```diff
@@ -122,24 +127,33 @@ define([
          * Sets the dropdown value to the given value. Constructs an id from it that (should)
          * be unique enough to apply to the dropdown.
          */
-        function setControlValue(value) {
+        function updateControlValue() {
+            const value = model.value;
+            const displayValue = model.displayValue || {};
             const control = ui.getElement('input-container.input');
-            if ($(control).find('option[value="' + value + '"]').length) {
-                $(control).val(value).trigger('change');
-            } else {
-                const newOption = new Option(value, value, true, true);
-                $(control).append(newOption).trigger('change');
+            if (model.value) {
+                const selectorValue = value.replace(/"/g, '\\"').replace(/'/g, "\\'");
+                if ($(control).find(`option[value="${selectorValue}"]`).length) {
+                    $(control).val(value).trigger('change');
+                } else {
+                    const dataAdapter = $(control).data('select2').dataAdapter;
+                    const newOption = dataAdapter.option(
+                        Object.assign({ id: model.value, value: model.value }, displayValue)
+                    );
+                    dataAdapter.addOptions(newOption);
+                    $(control).val(value).trigger('change');
```

Comment:
From some deep diving in Select2 code and docs, this creates a new option with all of the data that the Select2 option would otherwise see from a successful search, and lets Select2 do the usual rendering.